pennylane/ops/qubit.py

```diff
@@ -151,6 +152,9 @@ def adjoint(self, do_queue=False):
         return PauliX(wires=self.wires, do_queue=do_queue)
 
 
+register_control(PauliX, lambda op, wire: CNOT(wires=Wires(wire) + op.wires))
```

Comment:
Is there a more intuitive name, such as `controlled_version`?

Reply:
I called it `register_control` as I expect users making custom gates to have to use this. So when they make their custom gate, they can "register" what their controlled version decomposition is.

I'm not that tied to the name, so I could still changei.

Reply:
I want to start avoiding adding transforms directly on gates (and I think I'll change how adjoint is structured too). 

The reason why is because having `qml.RX(...).adjoint()` and a say `qml.RX.controlled_by(...)` would not do what what the user would initially expect. Having the transforms be registered separately from the class would prevent this confusion.

Reply:
I am not seeing `adjoint` and `controlled_by` as user-facing methods, but as a way to collect all data on a gate and store it in the class. That way, if a gate gets added or deleted, we do not have to update other obscure lists of information. No?

Reply:
My fear is that a user might _think_ they're user facing.

I think the compromise is just to call it `_controlled` instead. I'll do that.